contracts/balancer/IndexPool.sol

```diff
@@ -625,6 +627,7 @@ contract IndexPool is BToken, BMath, IIndexPool {
           uint256 additionalBalance = bsub(balance, minimumBalance);
           uint256 balRatio = bdiv(additionalBalance, minimumBalance);
           uint96 newDenorm = uint96(badd(MIN_WEIGHT, bmul(MIN_WEIGHT, balRatio)));
+          if (newDenorm > 2 * MIN_WEIGHT) newDenorm = uint96(2 * MIN_WEIGHT);
```

Comment:
Cutting off the result like this is safe?

Reply:
This adjustment of the weight on initialization isn't necessary, it's essentially just there to prevent tokens being overpriced when they are first introduced because they start out with a low weight.  This just restricts the extent to which the weight can be affected.